services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -3762,56 +3763,48 @@ private OfflinePushStatusInfo getOffLineJobStatus(
-          Version storeVersion = parentStore.getVersion(versionNum);
-          boolean isVersionPushed = storeVersion != null && storeVersion.getStatus().equals(PUSHED);
-          boolean isHybridStore = storeVersion != null && storeVersion.getHybridStoreConfig() != null;
-          boolean isTargetRegionPushWithDeferredSwapForCurrentVersion =
-              isTargetRegionPush && version.isVersionSwapDeferred();
-          // Truncate topic after push is in terminal state if
-          // 1. Its a hybrid store or regular push. (Hybrid store target push uses repush where isTargetRegionPush is
-          // false)
-          // 2. If target region push is enabled and job to push data only to target region completed (status == PUSHED)
-          // 3. If it is a target region push with deferred swap and a majority of regions have reached terminal status
-          if (!isTargetRegionPush // regular push
-              || isVersionPushed // target region push
-              || isHybridStore || isTargetRegionPushWithDeferredSwapForCurrentVersion) {
-            LOGGER
-                .info("Truncating parent VT {} after push status {}", kafkaTopic, currentReturnStatus.getRootStatus());
-            truncateTopicsOptionally(
-                clusterName,
-                kafkaTopic,
-                incrementalPushVersion,
-                currentReturnStatus,
-                currentReturnStatusDetails);
-          }
-          // status PUSHED is set when batch store's target region push is completed, but other region are yet to
-          // complete
-          if (currentReturnStatus.equals(ExecutionStatus.COMPLETED)) {
-            if (isTargetRegionPush && !isVersionPushed) {
-              parentStore.updateVersionStatus(versionNum, PUSHED);
-              repository.updateStore(parentStore);
-              LOGGER.info("Updating parent store version {} status to {}", kafkaTopic, PUSHED);
-            } else { // status ONLINE is set when all region finishes ingestion for either regular or target region
-                     // push.
-              parentStore.updateVersionStatus(versionNum, ONLINE);
-              repository.updateStore(parentStore);
-              LOGGER.info("Updating parent store version {} status to {}", kafkaTopic, ONLINE);
-            }
-          }
+          handleTerminalJobStatus(
+              clusterName,
+              kafkaTopic,
+              incrementalPushVersion,
+              targetedRegions,
+              parentStore,
+              version,
+              versionNum,
+              currentReturnStatus,
+              currentReturnStatusDetails,
+              repository);
+        }
+      } else {
+        // If the aggregate status is not terminal, but the parent version status is marked as KILLED, we should mark
+        // the
+        // push job status as terminal (ERROR) as job was killed
+        Version parentStoreVersion = parentStore.getVersion(versionNum);
```

Comment:
is it not same as `Version version = parentStore.getVersion(versionNum);` declared before ?